### PR TITLE
Respect the "sensitive" flag for posts

### PR DESCRIPTION
**Pull request by annando**
Posts on for example Mastodon can be marked as "sensitive" in different ways. One is when a "content warning" is entered, but there are other ways as well, for example you can define your post as being "sensitive" in the Mastodon API without providing any data for the "spoiler_text" field: https://docs.joinmastodon.org/methods/statuses/#form-data-parameters

This field is now respected, when we display posts.

With this change we don't need to blur any pictures any more, when the post is set to sensitive:

When viewed:
image

When expanded:
image

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -3440,8 +3439,8 @@ public static function prepareBody(array &$item, bool $attach = false, bool $is_
 				$filter_reasons[] = DI::l10n()->t('Content from %s is collapsed', $item['author-name']);
 			}
 
-			if (!empty($item['content-warning']) && (!$uid || !DI::pConfig()->get($uid, 'system', 'disable_cw', false))) {
-				$filter_reasons[] = DI::l10n()->t('Content warning: %s', $item['content-warning']);
+			if ($item['sensitive'] && (!$uid || !DI::pConfig()->get($uid, 'system', 'disable_cw', false))) {
```
Comment (MrPetovan) [inline src/Model/Item.php:3442]: Quid of the `system. display_sensitive` pconfig key?

**Review by MrPetovan — COMMENTED** [review]
Shouldn't we keep both `system.enable_cw` and `system.display_sensitive`?

**Comment by annando** [comment]
What should the other setting now control?

**Comment by MrPetovan** [comment]
- `system.enable_cw`: Collapses posts with a summary
- `system.display_sensitive`: Doesn't collapse sensitive posts

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php:3442]: ?

**Comment by annando** [comment]
There would be a third option as well: Collapse posts that marked as sensitive and that contain media. I would favour for possible a later PR introducing a new option that only collapses sensitive post when they contain media or a summary.

**Comment by MrPetovan** [comment]
You want a third option but you reduced the options to one in this PR, how about you leave the two current options for now until you introduce the third?

**Comment by annando** [comment]
`display_sensitive` was about blurring the images. I've removed the blurring. So that option doesn't make sense anymore. But I also wanted to simplify it for now, and then get some experience and feedback from the users, so that we can then find out what settings do make sense.

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**